src/ride/track.c

```diff
@@ -4921,15 +4922,445 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+	edx = (edx + 3) & 0x0F;
+
+	uint32 edx1 = edx;// loc_6CDC37:
+	while (true) {
+		// 0x6cdc37
+		mapElement->properties.track.maze_entry |= (1 << edx);
+
+		edx1 = edx;
+
+		uint32 direction = RCT2_GLOBAL(0x00993D0C + edx1, uint8);
+		uint16 flooredX = floor2(x, 32);
+		uint16 flooredY = floor2(y, 32);
+
+		flooredX -= TileDirectionDelta[direction].x;
+		flooredY -= TileDirectionDelta[direction].y;
+
+		rct_map_element *tmp_mapElement = map_get_first_element_at(flooredX / 32, flooredY / 32);
+
+		uint8 bl = (z / 8) & 0xFF;
+		uint8 bh = rideIndex;
+
+		do {
+			// loc_6CDC7E:
+			if (map_element_get_type(tmp_mapElement) != MAP_ELEMENT_TYPE_TRACK) {
+				continue;
+			}
+
+			if (tmp_mapElement->base_height != bl) {
+				continue;
+			}
+
+			if (tmp_mapElement->properties.track.ride_index != bh) {
+				continue;
+			}
+
+			if (tmp_mapElement->properties.track.type != 0x65) {
+				continue;
+			}
+
+			// loc_6CDCA5
+			uint8 edx11 = RCT2_GLOBAL(0x993CFC + edx1, uint8);
+			tmp_mapElement->properties.track.maze_entry |= 1 << (edx11 & 0xFFFF);
+
+			break;
+		} while (!map_element_is_last_for_tile(tmp_mapElement++));
+
+		edx--;
+
+		if ((edx & 0x3) == 0x3) {
+			break;
+		}
+	}
+
+	return loc_6CDDA2(originalX, originalY, rideIndex, mapElement);
+}
+
+money32 loc_6CDB35(uint16 x, uint8 flags, uint8 direction, uint16 y, uint8 rideIndex, uint8 mode, uint16 unknown, uint16 z, rct_map_element *mapElement) {
+	uint8 bl;
+
+	if ((flags & GAME_COMMAND_FLAG_APPLY) == 0) {
+		return loc_6CDDFD();
+	}
+
+
+	if (mode == 1) {
+		return loc_6CDDA2(x, y, rideIndex, mapElement);
+	}
+
+	if (mode == 0) {
+		uint8 edx = get_edx(x, y);
+
+		// 0x6cdcff
+		mapElement->properties.track.maze_entry &= ~(1 << edx);
+
+		if (direction == 4) {
+			return loc_6CDDA2(x, y, rideIndex, mapElement);
+		}
+
+		// 0x6cdd12
+		edx = RCT2_GLOBAL(0x993ce9 + (direction + edx), uint8);
+		// TODO: wth is edx?
+		mapElement->properties.track.maze_entry &= ~(1 << edx);
+
+		// SAVE edx + esi
+
+		// TODO: btr     [esi+5], dx
+		// (esi+5 = mapElement->properties.track.maze_entry)
+
+		uint32 temp_edx = RCT2_GLOBAL(0x993cfc + edx, uint8);
+		if ((temp_edx & 0xFF) == 0xFF) {
+			return loc_6CDDA2(x, y, rideIndex, mapElement);
+		}
+
+		uint16 xRounded = floor2(x, 32);
+		uint16 yRounded = floor2(y, 32);
+
+		uint32 ebx = direction;
+		xRounded -= TileDirectionDelta[direction].x;
+		yRounded -= TileDirectionDelta[direction].y;
+
+		uint8 bh = rideIndex;
+		bl = (uint8) ((z >> 3) & 0xFF);
+
+		rct_map_element *temp_esi = map_get_first_element_at(xRounded / 32, yRounded / 32);
+
+		// 0x6cdd71:
+		do {
+			if (map_element_get_type(temp_esi) != MAP_ELEMENT_TYPE_TRACK) {
+				continue;
+			}
+
+			if (temp_esi->base_height != bl) {
+				continue;
+			}
+
+			if (temp_esi->properties.track.ride_index != rideIndex) {
+				continue;
+			}
+
+			if (temp_esi->properties.track.type != 0x65) {
+				continue;
+			}
+
+			// loc_6CDD9B;
+			temp_esi->properties.track.maze_entry &= ~(1 << edx);
```

Comment:
That solved the last issue [known] with maze building/hedge removing. Now to find out why hedge restoring doesn't work.